### PR TITLE
shipit_static_analysis: Pass in --quiet into eslint instead of --no-i…

### LINKED ISSUE #788: shipit_static_analysis: reviewbot uses --no-ignore with eslint
**MozReview URL:** https://reviewboard.mozilla.org/r/214284/

**Problem:** I noticed that reviewbot uses --no-ignore with eslint:
https://github.com/mozilla-releng/services/blob/master/src/shipit_static_analysis/shipit_static_analysis/lint.py#L128

This results in reviews being posted to files that would otherwise be ignored when running eslint locally (such as the example linked above). This means developers could run ./mach lint locally, see no errors, push to review, and have them posted.

I think the reason this was initially added was because eslint generates a dummy 'warning' error when it ignores a file, which would probably trip up the reviewbot. I think we'll need to explicitly detect + ignore this dummy warning error, either in the reviewbot, or in tools/lint/eslint. It probably makes sense to ignore this from in-tree. I'll file a bug.

So steps to fix this will be:
1) Ignore the dummy warning when eslint ignores a file
2) Remove --no-ignore

### PR DESCRIPTION
**Pull request by ahal**
…gnore, fixes #788

### CODE REVIEW

**Comment by La0** [comment]
tl;dr: Works for me, i'm merging this.

Long version: When i first tested `mozlint` i quickly saw that some fiels were ignored, with a warning such as:
```
File ignored because of a matching ignore pattern. Use "--no-ignore" to override
```

As it affected quite some files, i decided to use `--no-ignore` so that the bot would behave as for `clang-tidy` execution where we have no paths checks (except for 3rd party later on).

I didn't look into `--quiet` but it works nicely here as it does not output the ignore warning (the bot does not have to check for this).
